### PR TITLE
Search for all settings in config_plugins

### PR DESCRIPTION
**Pull request by sarahjcotton**
Covers issue #93

### CODE REVIEW

File: cleaner/environment_matrix/classes/local/matrix.php [hunk 1]
```diff
@@ -117,6 +118,36 @@ public static function search($search, $configitems = []) {
 
         }
 
+        $likeplugin = $DB->sql_like('plugin', ':likeplugin');
```
Comment (brendanheywood) [inline cleaner/environment_matrix/classes/local/matrix.php:121]: This is good but we need to protect against searching for config which is set in config.php and which doesn’t have a setting as these are things like db config which should never be leaked

we probably also need to handle when the db is set but the value is overridden in config.php

Comment (sarahjcotton) [reply cleaner/environment_matrix/classes/local/matrix.php:121]: The search doesn't return anything sensitive from config.php. The existing search uses core admin tree functionality and the new bit only searches config_plugins:

Screenshot from 2023-10-23 09-12-53

As for config.php overrides, we are only adding the value we want in the other environment at this point, not actually setting it, and in any case even if we set a value in the DB, the config.php version overrides that so it still shouldn't be an issue.

I have added an extra check though to check to make sure any settings already in the result array aren't overwritten as they could have previously been identified as being a textarea or html setting and the additional search can't check for that.

**Comment by sarahjcotton** [comment]
I think this is ready for review now.

**Comment by brendanheywood** [comment]
ok all looks great thanks @sarahjcotton

**Comment by sarahjcotton** [comment]
Thanks @brendanheywood !

**Pull request merged**